Sources/MapboxCoreNavigation/RouteController.swift

```diff
@@ -237,9 +237,11 @@ open class RouteController: NSObject {
     private lazy var sharedNavigator: Navigator = {
         return Navigator.shared
     }()
-    
+
+    private var navigatorProvider: NativeNavigatorProvider.Type
```

Comment:
Why not do the same as `PassiveLocationManager` and use `NavigatorProtocol` here too? I assume this is what it was created for.

Reply:
This will also highlight all the points we need to cover under Protocol, and will be a huge step to turn our 'integration' tests into real unit tests.

Reply:
Replaced with the protocol